### PR TITLE
Bugfix/mdns thrash

### PR DESCRIPTION
**Pull request by ProofOfKeags**
The presumed issue here is that we over aggressively reload the entire set of mdns services. Every time an add or a remove is called we reset the entire set of mdns addresses. This can happen in situations where a container is started or stopped, so in crash loop scenarios it can happen quite often. This PR changes the behavior to not reset the entry group when services are added, but rather we add the service. When we remove a service, we don't actually destroy the CNAME record until we've accumulated a thousand dead ones, or embassyd itself is restarted.

### CODE REVIEW

File: appmgr/src/net/mdns.rs [hunk 1]
```diff
@@ -107,38 +108,43 @@ impl MdnsControllerInner {
...
-                    avahi_sys::AVAHI_DNS_TYPE_CNAME as u16,
-                    avahi_sys::AVAHI_DEFAULT_TTL,
-                    self.hostname.as_ptr().cast(),
-                    self.hostname.len(),
+                self.load_one(key);
+            }
+        }
+    }
+    fn load_one(&self, key: &TorSecretKeyV3) {
+        unsafe {
```
Comment (Blu-J) [inline appmgr/src/net/mdns.rs:116]: Doesn't unsafe require the function be annotated too (Safety)?
https://doc.rust-lang.org/std/keyword.unsafe.html

File: appmgr/src/net/mdns.rs [hunk 2]
```diff
@@ -210,18 +217,30 @@ impl MdnsControllerInner {
         pkg_id: &PackageId,
         interfaces: I,
     ) {
-        self.services.extend(
-            interfaces
-                .into_iter()
-                .map(|(interface_id, key)| ((pkg_id.clone(), interface_id), key)),
-        );
-        self.sync();
+        let mut diff = false;
```
Comment (Blu-J) [inline appmgr/src/net/mdns.rs:220]: Would this be better as an iter().any() combinator?

File: appmgr/src/net/mdns.rs (hunk 1 above)
Comment (ProofOfKeags) [reply appmgr/src/net/mdns.rs:116]: that would make every function in rust unsafe

File: appmgr/src/net/mdns.rs (hunk 2 above)
Comment (ProofOfKeags) [reply appmgr/src/net/mdns.rs:220]: I'd love to tbh. I haven't figured out the way to make that clear. I'll take a stab at it again.

**Pull request closed without merging**